jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Jetty.java

```diff
@@ -65,7 +65,7 @@ public class Jetty
             pkg.getImplementationVersion() != null)
             VERSION = pkg.getImplementationVersion();
         else
-            VERSION = System.getProperty("jetty.version", __buildProperties.getProperty("version", "10.0.z-SNAPSHOT"));
+            VERSION = System.getProperty("jetty.version", __buildProperties.getProperty("version", "12.0.z-SNAPSHOT"));
```

Comment:
Note that this (the `10.0.z` part) is really only possible if people repackage the Jetty classes incorrectly.
The actual version string comes from the `/org/eclipse/jetty/version/build.properties` which is created during build time.

Reply:
Also, before the code even checks the value of the `jetty.version` property in the `/org/eclipse/jetty/version/build.properties` file, it will use the `Implementation-Version` value in the `META-INF/MANIFEST.MF`.

Reply:
Frankly, if the code reaches this point, it should probably just be the string `(unknown-due-to-bad-packaging)`.